### PR TITLE
Inverse of zero quaternion

### PR DESCRIPTION
**Pull request by RohitJain3241**
This throws ValueError whenever asked for inverse of Quaternion(0,0,0,0) as well as in form of q.pow(-1).
Hope its useful.

### CODE REVIEW

**Comment by RohitJain3241** [comment]
Hope this proves useful. @asmeurer @gxyd

**Comment by asmeurer** [comment]
It seems reasonable, but what do you think @Upabjojr.

@RohitJain3241 can you add a test?

File: sympy/algebras/quaternion.py [hunk 1]
```diff
@@ -303,6 +303,8 @@ def normalize(self):
     def inverse(self):
         """Returns the inverse of the quaternion."""
         q = self
+        if not q.norm():
+            raise ValueError("Cannot compute inverse for Quaternion(0,0,0,0)")
```
Comment (jksuom) [inline sympy/algebras/quaternion.py]: Perhaps something like "Cannot compute inverse for a quaternion with zero norm".

Comment (RohitJain3241) [reply sympy/algebras/quaternion.py]: That would also be a good.

**Comment by RohitJain3241** [comment]
@asmeurer  Surely I would add tests for this. But, what about the case for Quaternion(oo,oo,oo,oo) ? Should I consider that too??

**Comment by gxyd** [comment]
The first commit of this PR is https://github.com/sympy/sympy/pull/13693/commits/ae663241c4a8c98b8f3e20de137fa481cbc12a30 , why do we have that?

That is not related to this PR.

**Comment by RohitJain3241** [comment]
@gxyd Is this correct now? I am new to git and would need guidance if not correct.

File: sympy/algebras/tests/test_quaternion.py [hunk 2]
```diff
@@ -12,8 +13,8 @@ def test_quaternion_construction():
     assert q + q == Quaternion(2*x, 2*y, 2*z, 2*w)
 
     q2 = Quaternion.from_axis_angle((sqrt(3)/3, sqrt(3)/3, sqrt(3)/3), 2*pi/3)
-    assert q2 == Quaternion(Rational(1/2), Rational(1/2),
-                            Rational(1/2), Rational(1,2))
+    assert q2 == Quaternion(Rational(1,2), Rational(1,2),
```
Comment (gxyd) [inline sympy/algebras/tests/test_quaternion.py]: This change is not related to this PR anyway, please don't make this change here.

File: sympy/algebras/tests/test_quaternion.py [hunk 3]
```diff
@@ -79,7 +84,7 @@ def test_quaternion_functions():
     assert integrate(Quaternion(x, x, x, x), x) == \
     Quaternion(x**2 / 2, x**2 / 2, x**2 / 2, x**2 / 2)
 
-    assert Quaternion.rotate_point((1, 1, 1), q1) == (1 / 5, 1, 7 / 5)
+    assert Quaternion.rotate_point((1, 1, 1), q1) == (Rational(1,5) , 1 , Rational(7,5))
```
Comment (gxyd) [inline sympy/algebras/tests/test_quaternion.py]: un-related change, don't make this change here.

File: sympy/algebras/tests/test_quaternion.py [hunk 4]
```diff
@@ -50,11 +51,15 @@ def test_quaternion_complex_real_addition():
...
     q1 = Quaternion(1, 2, 3, 4)
+    q0 = Quaternion(0, 0, 0, 0)
 
     assert conjugate(q) == Quaternion(x, -y, -z, -w)
     assert q.norm() == sqrt(w**2 + x**2 + y**2 + z**2)
     assert q.normalize() == Quaternion(x, y, z, w) / sqrt(w**2 + x**2 + y**2 + z**2)
     assert q.inverse() == Quaternion(x, -y, -z, -w) / (w**2 + x**2 + y**2 + z**2)
+    def test1(p):
+        p1 = p.inverse()
+    raises(ValueError, lambda: test1(q0))
```
Comment (gxyd) [inline sympy/algebras/tests/test_quaternion.py]: There should be no need to write a `test1` function to test this change, it can be tested without it.

**Comment by gxyd** [comment]
> And even if we consider that we get ( Quaternion(oo,oo,oo,oo).inverse() ).inverse() as nan which is incorrect.

Seems like there now is a separate for that #13700 now. Anyway you can now focus on the current issue.

File: sympy/algebras/tests/test_quaternion.py (hunk 3 above)
Comment (RohitJain3241) [reply sympy/algebras/tests/test_quaternion.py]: It would be good if I do that in a separate PR? @gxyd

**Comment by RohitJain3241** [comment]
@gxyd For this issue to work properly I need this to work first #13710. Is it good or I do it for this first?

File: sympy/algebras/tests/test_quaternion.py (hunk 3 above)
Comment (gxyd) [reply sympy/algebras/tests/test_quaternion.py]: Yes, you make a separate PR for any other changes, which are not related to the issue you are solving.

**Comment by gxyd** [comment]
You can complete this issue first.

**Review by gxyd — APPROVED** [review]
LGTM, let's wait for @Upabjojr if he has any comments.

**Comment by Upabjojr** [comment]
I'm +1

**Comment by gxyd** [comment]
Thanks @Upabjojr @jksuom for reviewing. Looks good merging, thanks for your contribution.
